File: tests/rosettacode/doors100.py
# 100 doors problem
doors = [False] * 101
for p in range(1, 101):
    for d in range(p, 101, p):
        doors[d] = not doors[d]

open_doors = [i for i in range(1, 101) if doors[i]]
print("Open doors:", " ".join(map(str, open_doors)))

File: tests/rosettacode/doors100.clj
;; 100 doors problem
(let [doors (reduce (fn [d pass]
                      (reduce #(update %1 %2 not)
                              d (range pass 101 pass)))
                    (vec (repeat 101 false))
                    (range 1 101))
      open (filter #(nth doors %) (range 1 101))]
  (println (str "Open doors: " (clojure.string/join " " open))))
